worker.go

```diff
@@ -710,3 +721,25 @@ func cniEventHandler(ctx context.Context, targetCNIcfg *aws.TargetCNIconfig,
 		}
 	}
 }
+
+func equalSlices[T comparable](a, b []T) bool {
```

Comment:
This looks like it compares for equality ignoring order (and duplicates in a).

I am wondering if we can pull in slices helper lib and use https://pkg.go.dev/golang.org/x/exp/slices#Equal 
I think it will be part of stdlib on the next release, see https://github.com/golang/go/issues/57433 and https://pkg.go.dev/slices#Equal

If order should be ignored we can presort them.

Reply:
Or maybe just rename it to hint that it ignores element order

Reply:
I think its fine: $O(n^2)$ does not matter assuming slices are small and it does not allocate but the name could be misleading.

Reply:
Changed to `slices.Equal` plus sorting first. `n` will be at most `3` so that is not a concern :)